### DISCUSSION TITLE
Embedding Neovim as an extension/plugin for Zed

### DISCUSSION THREAD

**Discussion by chrisintheshell** [zed-industries/zed#17500] [Feature Requests]
Good day,

I'm curious if you looked into embedding Neovim as an extension/plugin for Zed to use Zed as a GUI for Neovim to get all the Vi/Vim functionality? Similar to the VSCode extension vscode-neovim (https://github.com/vscode-neovim/vscode-neovim).

This way would you wouldn't have to write code to mimic Vi/Vim commands but use a native implementation instead.

Love the Zed editor though. Thanks for all that you do.

-Chris

**Comment by bajrangCoder** [comment]
First, Watch this official video regarding embedding neovim inside Zed : [click here](https://youtu.be/Ys8-KkzH5Rc?si=nZjDjkvcAsSOUmfE)

And second thing, these types of extensions aren't supported yet

**Reply by chrisintheshell** [reply]
Thanks for this. Haven't watched the whole thing yet but it makes perfect sense.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
